src/doom/p_mobj.c

```diff
@@ -803,6 +803,12 @@ void P_SpawnMapThing (mapthing_t* mthing)
 	bit = 1;
     else if (gameskill == sk_nightmare)
 	bit = 4;
+    else if (gameskill < sk_baby)
+        // avoid undefined behavior (left shift by negative value and rhs too big)
+        // by accurately emulating what doom.exe did: select lower 8 bits and
```

Comment:
This comment is not accurate. Leave out "select lower 8 bits" since it doesn't do that.

Reply:
Very. It comes from the semantics of the assembly instruction. There's a link in the original bug report.

Reply:
> How certain are we about the `0x1F` part?

I changed [the sample code I created](https://github.com/chocolate-doom/chocolate-doom/issues/1677#issuecomment-2053722925) to use the `0x1F` constant and it seems to emulate perfectly the behavior of negative shift using Watcom compiler and DOSBox.